packages/react-dom/src/client/ReactDOMFiberComponent.js

```diff
@@ -810,6 +810,13 @@ export function updateProperties(
   lastRawProps: Object,
   nextRawProps: Object,
 ): void {
+  // Update checked *before* name.
+  // In the middle of an update, it is possible to have multiple checked.
+  // When a checked radio tries to change name, browser makes another radio's checked false.
+  if (tag === 'input') {
+    ReactDOMFiberInput.updateChecked(domElement, nextRawProps, 'radio');
+  }
```

Comment:
I think updateNamedCousins is called only on the button because of this code `ReactTestUtils.Simulate.click(buttonNode);`
I checked that updateNamedCousins is not called when I run your fixture.